Add option $showAllErrors to Html::errorSummary()

PowerGamer1 opened the pull request:
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | https://github.com/yiisoft/yii2/issues/12193 |

PowerGamer1 committed:
Update BaseHtml.php

Added ability to skip duplicate error messages and show all error messages in `errorSummary`.

PowerGamer1 committed:
Update BaseHtml.php

PowerGamer1 committed:
Update BaseHtml.php

PowerGamer1 committed:
Update CHANGELOG.md

SilverFire commented:
We use 4 spaces instead of `tab`

SilverFire commented on framework/helpers/BaseHtml.php:
```diff
@@ -1154,6 +1154,9 @@ public static function activeHint($model, $attribute, $options = [])
      * - footer: string, the footer HTML for the error summary.
      * - encode: boolean, if set to false then the error messages won't be encoded.
+     * @since 2.0.10
+     * - showAllErrors: boolean, if set to true every error message for each attribute will be shown otherwise
```
```
- showAllErrors: boolean, if set to true, all error messages for all the attributes will be shown. Otherwise only the first error message for each attribute will be shown. Option is available since 2.0.10
```

The previous line should be removed

SilverFire commented:
How about a unit test?

PowerGamer1 committed:
Update BaseHtml.php

Cosmetic changes.

PowerGamer1 commented:
@SilverFire 
I fixed tabs and removed "since" comment but I can't create tests since I never done unit testing and don't really know how to do it properly. It would be nice if you or someone else can decide what and how should be tested and create the tests.

SilverFire commented:
Ok, I'll take care of test a little bit later. Thank you

SilverFire commented:
Merged https://github.com/yiisoft/yii2/commit/bb7a36c5c682529f52bdea94fe940095f9463993

SilverFire commented:
Thank you!
